### ISSUE TITLE
Custom telemetry does not arrive at Application Insight

### ISSUE THREAD

**Issue by AndersDyhr** [Azure/azure-functions-host#7688]
#### Check for a solution in the Azure portal
Done, not applicable 

#### Issue:
I have made an Azure function that send custom telemetry to Application Insight when the function is run locally. However, 
 running the function one azure function host custom telemetry from the Java Azure function is not being send to Application Insight.

#### Please provide the following:

- Timestamp: 2021-09-23T06:50:24Z
- Function App version: 3.2 (Linux/Java) (Same observation on windows)
- Function App name: anderstest-functions-20210922205930008
- Function name: HttpExample
- Invocation ID: 33e8d4c7-6f66-4d58-82bf-7213ed632eaf
- Region: westeurope

#### Repro steps

1.	Build and deploy attached project to Azure in bash (anderstest-functions.zip)
                       a.	az login 
                       b.	go to project folder
                      c.	mvn clean package
                     d.	mvn azure-function:deploy
2.	Go to Azure portal and copy Instrumentation key for the newly created Application Insight in the function resource group
3.	In the Function.java file replace <INSTRUMENTATION KEY> with the copied key
4.	Re-deploy the function
    a.	     mvn clean package
    b.	     mvn azure-function:deploy
5.	As part of clean package a test is run locally on the computer, which produces a customEvent in the Application insights transaction search called “WinGame” after a short wait (as expected)
6.	Go to the created function app and into the function HttpExample
7.	Go to Code + Test tab
8.	Press Test/Run and choose any input and press Run
Confirm that log entries from the context logger appears in Application Insights after a short wait, but that the custom event “WinGame” never appears. 

#### Expected behavior

Custom event telemetry should appear in Application Insight 

#### Actual behavior

Custom event telemetry does not appear in Application Insight when run from azure (However, does when run locally)

#### Related information 

Provide any related information 

Fucntion code:

```java
public class Function {
    static final TelemetryClient telemetryClient = new TelemetryClient();

    @FunctionName("HttpExample")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET, HttpMethod.POST},
                authLevel = AuthorizationLevel.ANONYMOUS)
                HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {
        context.getLogger().warning("Java HTTP trigger processed a request.");

        context.getLogger().warning("Sending custom telemetry ");

        telemetryClient.getContext().setInstrumentationKey("<INSTRUMENTATION KEY>");
        telemetryClient.trackEvent("WinGame");

        telemetryClient.flush();

        context.getLogger().warning("Custom telemetry send");


        // Parse query parameter
        final String query = request.getQueryParameters().get("name");
        final String name = request.getBody().orElse(query);

        if (name == null) {
            return request.createResponseBuilder(HttpStatus.BAD_REQUEST).body("Please pass a name on the query string or in the request body").build();
        } else {
            return request.createResponseBuilder(HttpStatus.OK).body("Hello, " + name).build();
        }
    }
}
```
[anderstest-functions.zip](https://github.com/Azure/azure-functions-host/files/7215906/anderstest-functions.zip)

**Comment by v-bbalaiagar** [comment]
Hi @AndersDyhr, Thank you for your feedback! We will check for the possibilities internally and update you with the findings.
Meanwhile, here is a [blog](https://jonathancrozier.com/blog/recording-custom-telemetry-with-azure-application-insights) post which addresses similar scenario where the blog talks about custom telemetry.
Please check and let us know if this helps.

**Comment by AndersDyhr** [comment]
Thanks for that looking forward to hear back from you. The blog does not seem to help.

**Comment by v-bbalaiagar** [comment]
Hi @AndersDyhr,  Thank you for your feedback! We will check for the possibilities internally and update you with the findings.

**Comment by AndersDyhr** [comment]
Do you have any findings?

**Comment by v-bbalaiagar** [comment]
Hi @AndersDyhr, I checked this internally. Team wants to investigate this issue further and update you with the findings.

**Comment by AndersDyhr** [comment]
ok, thanks for the status

**Comment by AndersDyhr** [comment]
haven't heard back from you, are there any status on the investigation?

**Comment by amamounelsayed** [comment]
Thank you for your patience, For linux could you please make sure that you are using these app settings, https://docs.microsoft.com/en-us/azure/azure-monitor/app/monitor-functions ?

Thank you so much

**Comment by AndersDyhr** [comment]
we are using these exact settings, we also see the same issue on windows as well

**Comment by v-bbalaiagar** [comment]
Hi @amamounelsayed, Could you please look at this issue here

**Comment by AndersDyhr** [comment]
Any progress?

**Comment by amamounelsayed** [comment]
@AndersDyhr is this the same app anderstest-functions-20210922205930008?
I tried to check it but I believe it got deleted. Do you have any other application that faces the same issue?

Thank you,
Ahmed

**Comment by AndersDyhr** [comment]
yes it has been deleted. We don't have other that uses it since it doesn't work at the moment. I've written step by step guide how to reproduce the issue, so you can easily try it for your self?

**Comment by atdi** [comment]
I have the same issue and is not working regardless what I am trying to do. When I am sending custom telemetry from my local machine is working, but when the function is running in cloud it doesn't arrive to app insigths.
